src/main/java/org/springframework/data/redis/cache/DefaultRedisCacheWriter.java

```diff
@@ -124,12 +124,14 @@ public void put(String name, byte[] key, byte[] value, @Nullable Duration ttl) {
 	}
 
 	@Override
-	public byte[] get(String name, byte[] key) {
+	public byte[] get(String name, byte[] key, @Nullable Duration ttl) {
 
 		Assert.notNull(name, "Name must not be null");
 		Assert.notNull(key, "Key must not be null");
 
-		byte[] result = execute(name, connection -> connection.get(key));
+		byte[] result = shouldExpireWithin(ttl)
+			? execute(name, connection -> connection.getEx(key, Expiration.from(ttl)))
```

Comment:
Should we fallback to `get` and `expire` if server doesn't support `getex`?

Reply:
Thank you for the feedback, @quaff.

I spoke with @mp911de about this earlier today and we decided this feature will only be included from Spring Data Redis `3.2` and later, which is based on the latest Jedis and Lettuce drivers supporting most of the features in the newest Redis server releases.

In addition, it is an opt-in feature. See [here](https://github.com/spring-projects/spring-data-redis/pull/2649/files#diff-9b091dbe458d10f65774fb0a47e282e73b8535d4d9ae178342041badf6fb3e46R50) and [here](https://github.com/spring-projects/spring-data-redis/pull/2649/files#diff-9b091dbe458d10f65774fb0a47e282e73b8535d4d9ae178342041badf6fb3e46R225-R228), defaulting to the previous behavior.

Reply:
> which is based on the latest Jedis and Lettuce drivers supporting most of the features in the newest Redis server releases.

I mean server not client doesn't support `getex`, it's a new command since redis 6.2.0, upgrading server is not convenient as client, many server still using 4.x and 5.x, especially service provided by private cloud.

Reply:
If the server does not support `getex` then you cannot use this feature.

Reply:
We could fallback to `get` and `expire`.

Reply:
@quaff - I understood what you meant between client and server.

@christophstrobl and I are meeting this week on the topic of caching and using TTI-like expiration in SD Redis. Let me touch base with **Christoph** on this whether or not we will take into consideration Redis server versions.